### PR TITLE
refactor(tools): consolidate tool-arg helpers into internal/tools/toolargs

### PR DESCRIPTION
**Pull request by thane-developer**
## Summary

Refs #965. Three tool packages each carried prefixed copies of the same JSON-arg extraction logic — `stringArg`/`ldStringArg`/`mrStringArg`, `intArg`/`ldIntArg`/`mrIntArg`, etc. — prefixed *only* because they'd collide. The prefixes were a workaround for the missing shared home; subtle coercion differences between the copies were a standing drift risk (the issue's core smell).

This lands the canonical home and migrates the `internal/tools` package onto it.

### New package: `internal/tools/toolargs`

Full translator API with thorough table-driven tests:

- `String` / `TrimmedString`
- `Int` / `IntOr` / `IntOK`
- `Bool` / `BoolOr` / `BoolOK` / `HasBool`
- `StringSlice` / `Uint32Slice`

Tests cover the JSON-coercion edge cases the issue flagged: `float64`/`json.Number` integrality, numeric strings, `"true"`/`"false"` string bools, the #930 omitted-bool default, and uint32 range/fractional filtering.

### Migration

Deleted the two prefixed sets in `internal/tools` (`mr*` in `model_registry_tool_helpers.go`, `ld*` string/int in `loop_definition_tool_helpers.go`) and repointed ~71 call sites to `toolargs`. Loop-specific decoders (`decodeLoopSpecArg`, `decodeLoopLaunchArg`) and registry-specific helpers stay put.

## Decisions worth flagging

- **Package named `toolargs`, not `args`** (the issue's sketch). Handlers universally name their argument map `args map[string]any`, so a package named `args` would be shadowed at nearly every call site — `args.String(args, key)` won't compile. `toolargs.String(args, key)` reads cleanly with no alias.
- **Canonical `Int` rejects fractional floats** (the model-registry semantics) rather than truncating (older email/loop behavior). Migrated call sites only ever pass whole-number args, so behavior is unchanged in practice; this picks the stricter, more correct contract as the single source of truth.
- **Architecture baseline:** packages 58 → 59 for the new leaf package (updated in `scripts/architecture.baseline`).

## Out of scope (follow-ups)

- **`internal/channels/email`** helper set — it carries unique UID coercion (`uint32SliceArg`/`coerceUint32`), the #930 bool-default, and warn-logging worth migrating carefully. The new `toolargs` API already includes `Uint32Slice`/`StringSlice`/`BoolOr` so that migration is pure call-site repointing.
- **~167 inline `args[...].(T)` assertion sites** — the issue's step 3, explicitly a separate opportunistic sweep.

This PR therefore uses `Refs #965`, not `Closes`.

## Test plan

- [x] `go test ./internal/tools/...` (incl. new `toolargs` package)
- [x] `just ci` green locally
- [x] Architecture baseline updated and passing
- [x] No unrelated import-grouping churn (reverted goimports `-local` side effects on untouched files)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: internal/tools/toolargs/toolargs.go [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+		if v != float64(int(v)) {
+			return 0, false
+		}
+		return int(v), true
+	case json.Number:
+		n, err := v.Int64()
+		if err != nil {
+			return 0, false
+		}
+		return int(n), true
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/tools/toolargs/toolargs.go]: Fixed in 068f3e6. `IntOK`'s json.Number branch now falls back to `Float64()` with the same integrality check as the float64 case, so decimal-form integral values like `9.0` coerce while `9.5` is still rejected. Added a table case (`json.Number("9.0") → 9, true`).

File: internal/tools/toolargs/toolargs_test.go [hunk 2]
```diff
@@ -0,0 +1,211 @@
+package toolargs
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+)
+
+func TestString(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/tools/toolargs/toolargs_test.go:9]: Fixed in 068f3e6. Added runnable `Example*` tests (ExampleString/Int/BoolOr/StringSlice) covering the primary usage patterns, each with an `// Output:` block so `go test` verifies them — per AGENTS.md's exported-API example requirement.
